File: python-algorithm/leetcode/problem_191.py
"""191. Number of 1 Bits
https://leetcode.com/problems/number-of-1-bits/
"""


class Solution:
    def hammingWeight(self, n: int) -> int:
        def low_bit(x: int) -> int:
            return x & -x

        ans = 0
        while n != 0:
            n -= low_bit(n)
            ans += 1
        return ans

File: java-algorithm/src/main/java/com/leetcode/Problem191.java
package com.leetcode;

/**
 * 191. Number of 1 Bits
 * https://leetcode.com/problems/number-of-1-bits/
 */
public class Problem191 {
    public int hammingWeight(int n) {
        int ans = 0;
        for (int i = 0; i < 32; i++) {
            if ((n & (1 << i)) != 0) {
                ans++;
            }
        }
        return ans;
    }
}
